### PR TITLE
US007_Sort_Items

### PR DESCRIPTION
**Pull request by Zeroya**

### CODE REVIEW

File: src/store/reducers/UserSlice.ts [hunk 1]
```diff
@@ -79,9 +79,19 @@ export const counterSlice = createSlice({
...
         (state.filtValue = action.payload);
     },
+    sortTodosBy: (state, action: PayloadAction<string>) => {
+      SortOptions.message === action.payload &&
+        (state.todos = state.todos.sort((a: ITodo, b: ITodo) => (a.message > b.message ? 1 : -1)));
+      SortOptions.date === action.payload &&
+        (state.todos = state.todos.sort((a: ITodo, b: ITodo) => {
+          let dateA = new Date(getCreationModalDate(a.dateExpiration)).getTime();
+          let dateB = new Date(getCreationModalDate(b.dateExpiration)).getTime();
+          return dateA > dateB ? 1 : -1;
```
Comment (Yuliia143) [inline src/store/reducers/UserSlice.ts]: Try dateA - dateB

File: src/store/reducers/UserSlice.ts [hunk 2]
```diff
@@ -79,9 +79,19 @@ export const counterSlice = createSlice({
       (action.payload === Сondition.active || Сondition.completed || Сondition.all) &&
         (state.filtValue = action.payload);
     },
+    sortTodosBy: (state, action: PayloadAction<string>) => {
+      SortOptions.message === action.payload &&
+        (state.todos = state.todos.sort((a: ITodo, b: ITodo) => (a.message > b.message ? 1 : -1)));
```
Comment (Yuliia143) [inline src/store/reducers/UserSlice.ts]: Compare your messages using localeCompare

File: src/store/reducers/UserSlice.ts (hunk 1 above)
Comment (Zeroya) [reply src/store/reducers/UserSlice.ts]: Fixed

File: src/store/reducers/UserSlice.ts (hunk 2 above)
Comment (Zeroya) [reply src/store/reducers/UserSlice.ts]: Added

Comment (Zeroya) [reply src/store/reducers/UserSlice.ts:84]: I rewrite in slice sort, but think about need or not to rewrite with localeCompare in all project ?

Comment (Zeroya) [reply src/store/reducers/UserSlice.ts:84]: Because I see in some moments it looks more complicated

Comment (Zeroya) [reply src/store/reducers/UserSlice.ts:84]: Added changes to the whole project

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**